codegen-serde/src/test/kotlin/software/amazon/smithy/rust/codegen/serde/SerdeDecoratorTest.kt

```diff
@@ -0,0 +1,302 @@
+                        *codegenScope,
+                    )
+                }
+
+                unitTest("delegated_serde") {
+                    rustTemplate(
+                        """
+                        use #{crate}::input::SayHelloInput;
+                        use #{crate}::serde::*;
+                        ##[derive(#{serde}::Serialize)]
+                        struct MyRecord {
+                            ##[serde(serialize_with = "serialize_redacted")]
+                            redact_field: SayHelloInput,
+                            ##[serde(serialize_with = "serialize_unredacted")]
+                            unredacted_field: SayHelloInput
+                        }
+                        let input = SayHelloInput::builder().foo(Some("foo-value".to_string())).build().unwrap();
+
+                        let field = MyRecord {
+                            redact_field: input.clone(),
+                            unredacted_field: input
+                        };
+                        let serialized = #{serde_json}::to_string(&field).expect("failed to serialize");
+                        assert_eq!(serialized, r##"{"redact_field":{"foo":"<redacted>"},"unredacted_field":{"foo":"foo-value"}}"##);
+                        """,
+                        *codegenScope,
+                    )
+                }
+            }
+        }
+    }
+
+    private val expectedNoRedactions =
+        """{
+        "foo": "foo-value",
+        "e": "A",
+        "nested": {
+          "int": 5,
+          "sensitive": {
+            "a": "1970-01-01T00:00:00Z"
+          },
+          "notSensitive": {
+            "a": "1970-01-01T00:00:00Z"
+          },
+          "manyEnums": [
+            "A"
+          ],
+          "sparse": [null, "A", "B"]
+        },
+        "union": {
+          "enum": "B"
+        },
+        "document": "hello!",
+        "blob": "aGVsbG8="
+    }""".replace("\\s".toRegex(), "")
+
+    private val expectedRedacted =
+        """{
+        "foo": "<redacted>",
+        "e": "<redacted>",
+        "nested": {
+          "int": 5,
+          "sensitive": {
+            "a": "<redacted>"
+          },
+          "notSensitive": {
+            "a": "1970-01-01T00:00:00Z"
+          },
+          "manyEnums": [
+            "<redacted>"
+          ],
+          "sparse": ["<redacted>", "<redacted>", "<redacted>"]
+        },
+        "union": "<redacted>",
+        "document": "hello!",
+        "blob": "<redacted>"
+        }
+        """.replace("\\s".toRegex(), "")
+
+    @Test
+    fun generateSerializersThatWorkClient() {
+        clientIntegrationTest(simpleModel, params = IntegrationTestParams(cargoCommand = "cargo test --all-features")) { ctx, crate ->
+            val codegenScope =
+                arrayOf(
+                    "crate" to RustType.Opaque(ctx.moduleUseName()),
+                    "serde_json" to CargoDependency("serde_json", CratesIo("1")).toDevDependency().toType(),
```

Comment:
Can we also test with `serde_cbor` for good measure? To also check that Smithy blobs get serialized as bytes and not base64-encoded.

Reply:
done